server/src/main.rs

```diff
@@ -183,48 +183,54 @@ async fn handle_socket(socket: WebSocket, state: Arc<StdMutex<State>>) {
                 let _ = local_sender.send(Response::ServerTime(SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_millis() as u64)).await;
             },
             Request::JoinRoom(room_name, player_name, x, y, z) => {
-                log::info!("Player {player_id:?} ({player_name}) joins room {room_name:?}");
+                if room_name.len() > 128 {
+                    log::error!("Room name is greater than 128 chars.");
+                    local_sender.send(Response::RoomNameTooLong).await;
+                }
```

Comment:
Use a guard-if.

Reply:
I believe I cannot do so. Doing so would make the response be `RoomNameTooLong` then immediately change to `ReceiveFailed`. I have tested it and I believe the guard-if makes it not work as intended.

Reply:
Take a look at what the other guard-ifs in that function are doing.

Reply:
Oh my god I had to use `continue`.

Thanks :)